### PR TITLE
Add support for OS_TOKEN

### PR DESCRIPTION
**Pull request by jtopjian**
For #271

### CODE REVIEW

File: openstack/provider.go [hunk 1]
```diff
@@ -70,9 +70,12 @@ func Provider() terraform.ResourceProvider {
 			},
 
 			"token": &schema.Schema{
-				Type:        schema.TypeString,
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("OS_AUTH_TOKEN", ""),
+				Type:     schema.TypeString,
```
[the remark this answers is not in the document]
Comment (fatmcgav) [reply openstack/provider.go]: Fair nuf :) And yeh, now you say that I can see it further up this file... :)

**Comment by jtopjian** [comment]
recheck

**Comment by jtopjian** [comment]
recheck

**Comment by jtopjian** [comment]
The test failure has nothing to do with this PR. It's a test that occasionally fails for unknown reasons. However, I'm being stubborn and want to see it pass first.

**Comment by jtopjian** [comment]
recheck

**Comment by jtopjian** [comment]
@fatmcgav Thanks for the quick review on this one :)

**Pull request merged**